src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/StringPrefixTests.cs

```diff
@@ -0,0 +1,371 @@
+                .Name("Query")
+                .Field("test")
+                .UseFiltering<FooFilterType>()
+                .UseTestReport(Client)
+                .ResolveTestData(Client, _data))
+            .AddElasticSearchFiltering()
+            .BuildTestExecutorAsync();
+
+        const string query = @"
+        {
+            test(where: {qux: { startsWith: ""do"" }, bar: { startsWith: ""th"" }}) {
+                bar
+            }
+        }
+        ";
+
+        IExecutionResult result = await executorAsync.ExecuteAsync(query);
+        result.MatchQuerySnapshot();
+    }
+
+    [Fact]
+    public async Task ElasticSearch_MultipleField_OneNegated()
+    {
+        await IndexDocuments(_data);
+
+        IRequestExecutor executorAsync = await new ServiceCollection()
+            .AddGraphQL()
+            .AddQueryType(x => x
+                .Name("Query")
+                .Field("test")
+                .UseFiltering<FooFilterType>()
+                .UseTestReport(Client)
+                .ResolveTestData(Client, _data))
+            .AddElasticSearchFiltering()
+            .BuildTestExecutorAsync();
+
+        const string query = @"
+        {
+            test(where: {qux: { startsWith: ""do"" }, bar: { nstartsWith: ""th"" }}) {
+                bar
+            }
+        }
+        ";
+
+        IExecutionResult result = await executorAsync.ExecuteAsync(query);
+        result.MatchQuerySnapshot();
+    }
+
+    [Fact]
+    public async Task ElasticSearch_AndField()
+    {
+        await IndexDocuments(_data);
+
+        IRequestExecutor executorAsync = await new ServiceCollection()
+            .AddGraphQL()
+            .AddQueryType(x => x
+                .Name("Query")
+                .Field("test")
+                .UseFiltering<FooFilterType>()
+                .UseTestReport(Client)
+                .ResolveTestData(Client, _data))
+            .AddElasticSearchFiltering()
+            .BuildTestExecutorAsync();
+
+        const string query = @"
+        {
+            test(where: {and: [{bar: { startsWith: ""th"" }},{qux: { startsWith: ""do"" }}]}) {
+                bar
+            }
+        }
+        ";
+
+        IExecutionResult result = await executorAsync.ExecuteAsync(query);
+        result.MatchQuerySnapshot();
+    }
+
+    [Fact]
+    public async Task ElasticSearch_AndField_WithNegation()
+    {
+        await IndexDocuments(_data);
+
+        IRequestExecutor executorAsync = await new ServiceCollection()
+            .AddGraphQL()
+            .AddQueryType(x => x
+                .Name("Query")
+                .Field("test")
+                .UseFiltering<FooFilterType>()
+                .UseTestReport(Client)
+                .ResolveTestData(Client, _data))
+            .AddElasticSearchFiltering()
+            .BuildTestExecutorAsync();
+
+        const string query = @"
+        {
+            test(where: {and: [{bar: { startsWith: ""th"" }},{qux: { nstartsWith: ""do"" }}]}) {
```

Comment:
Can you check that this lines are not longer than 100 lines?

Reply:
Could you elaborate? What exactly needs to be less than 100 lines? the test file you mean?

Reply:
Thanks for letting me know about this policy. Luckily every line were less than 100 characters already, but I made some line breaks here and there to ensure this and increase readability.